sherpa/plot/backend_utils.py

```diff
@@ -0,0 +1,209 @@
+#
+#  Copyright (C) 2022
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+'''This module has utilities for backends.
+
+They are not useful on their own, but can help to define new plotting backends.
+'''
+
+import functools
+from inspect import signature
+import re
+
+__all__ = ('translate_args', 'add_kwargs_to_doc', 'get_keyword_defaults')
+
+kwargs_indent = re.compile(r"\n(?P<spaces>\s*){kwargs}",flags=re.M)
+
+def find_indent(doc):
+    match = kwargs_indent.findall(doc)
+    return len(match[0])
+
+
+def translate_args(func):
+    '''A decorator to translate function arguments.
+
+    When a method decorated with this decorator is called,
+    the decorator inspects the arguments that are passed into the
+    function. For each argument that is found as a key in the object's
+    ``translate_args`` dictionary, the value of that argument is
+    translated. The items in that dictionary can be functions or
+    dictionaries.
+
+    The purpose of this decorator is to support backends that use different
+    syntax for the same option, e.g. one backend might call a color
+    "red", while another uses the tuple (1, 0, 0) to describe the same color.
+
+    Example
+    -------
+    In this example, the input 'r' or 'b' will be translated into an rgb tuple
+    before the ``plot`` function is called. Other values (e.g. ``color=(0, 0, 0)``)
+    will be passed through unchanged so that the user can also make use of any other
+    color specification that the backend allows.
+
+    >>> class Plotter():
```

Comment:
```suggestion
    >>> class Plotter:
```

Not the most impressive of suggestions, I admit.

Reply:
But made me learn something.